src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -176,7 +182,12 @@ protected ExtensionsRunner(Extension extension) throws IOException {
             .put(TransportSettings.BIND_HOST.getKey(), extensionSettings.getHostAddress())
             .put(TransportSettings.PORT.getKey(), extensionSettings.getHostPort())
             .build();
-        this.threadPool = new ThreadPool(settings);
+
+        final List<ExecutorBuilder<?>> executorBuilders = extension.getExecutorBuilders(settings);
+
+        runnableTaskListener = new AtomicReference<>();
+        this.threadPool = new ThreadPool(settings, runnableTaskListener, executorBuilders.toArray(new ExecutorBuilder[0]));
```

Comment:
Instead of passing it like `new ExecutorBuilder[0]`, create a global variable above and pass it here

Reply:
What exactly `runnableTaskListener` is doing? I see it is just initialized and passed

Reply:
I have just emulated what is done in Node.java [here](https://github.com/opensearch-project/OpenSearch/blob/5eed61e263d9bdfbd64b2b677d4ef6933085dac5/server/src/main/java/org/opensearch/node/Node.java#L482) to add the executor builders to the thread pool

Reply:
> Instead of passing it like `new ExecutorBuilder[0]`, create a global variable above and pass it here

The format @joshpalis has is standard for creating an array from a list.  (JDK 8+ has a fancier format availbale, `ExecutorBuilder[]::new` but it's essentially the same.)
 - The list is required by the implemented extension point
 - The array is required by the varags `ThreadPool` constructor.

Reply:
> What exactly `runnableTaskListener` is doing? I see it is just initialized and passed

Looks like it's a reference to a [TaskResourceTrackingService](https://github.com/opensearch-project/OpenSearch/blob/37b37c5b518c5bdb2dfdc5250117cfd39394edeb/server/src/main/java/org/opensearch/node/Node.java#L1224) object which in turn looks like it tracks thread memory allocations.

Reply:
> Instead of passing it like `new ExecutorBuilder[0]`, create a global variable above and pass it here

@owaiskazi19 and @joshpalis you may be interested in why we use `[0]` here: https://shipilev.net/blog/2016/arrays-wisdom-ancients/